packages/diagrams/frontend/sirius-components-diagrams/src/renderer/actions/useActions.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  GQLAction,
+  GQLDiagramDescription,
+  GQLErrorPayload,
+  GQLGetActionsData,
+  GQLGetActionsVariables,
+  GQLInvokeActionData,
+  GQLInvokeActionInput,
+  GQLInvokeActionPayload,
+  GQLInvokeActionSuccessPayload,
+  GQLInvokeActionVariables,
+  GQLRepresentationDescription,
+  UseActionsProps,
+} from './useActions.types';
+
+export const getActionsQuery = gql`
+  query getActions($editingContextId: ID!, $diagramId: ID!, $diagramElementId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        representation(representationId: $diagramId) {
+          description {
+            ... on DiagramDescription {
+              nodeDescriptions {
+                id
+                actions(diagramElementId: $diagramElementId) {
+                  id
+                  label
+                  iconURL
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+const invokeActionMutation = gql`
+  mutation invokeAction($input: InvokeActionInput!) {
+    invokeAction(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isDiagramDescription = (
+  representationDescription: GQLRepresentationDescription
+): representationDescription is GQLDiagramDescription => representationDescription.__typename === 'DiagramDescription';
+
+const isErrorPayload = (payload: GQLInvokeActionPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isInvokeActionSuccessPayload = (payload: GQLInvokeActionPayload): payload is GQLInvokeActionSuccessPayload =>
+  payload.__typename === 'InvokeActionSuccessPayload';
+
+export const useActions = ({ diagramElementId, targetElementId, nodeDescriptionId }: UseActionsProps) => {
+  const { addErrorMessage, addMessages } = useMultiToast();
+  const { diagramId, editingContextId } = useContext<DiagramContextValue>(DiagramContext);
+
+  const { data: actionsData, error: actionsError } = useQuery<GQLGetActionsData, GQLGetActionsVariables>(
+    getActionsQuery,
+    {
+      variables: {
+        editingContextId,
+        diagramId,
+        diagramElementId,
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (actionsError) {
+      addErrorMessage('An unexpected error has occurred while retrieving actions, please refresh the page');
+    }
+  }, [actionsError]);
+
+  const description: GQLRepresentationDescription | undefined =
+    actionsData?.viewer.editingContext.representation.description;
+
+  const actions: GQLAction[] | null =
+    description && isDiagramDescription(description)
+      ? description.nodeDescriptions.filter((nd) => nd.id == nodeDescriptionId).flatMap((nd) => nd.actions)
+      : null;
+
+  const [invokeActionFrom] = useMutation<GQLInvokeActionData, GQLInvokeActionVariables>(invokeActionMutation);
+
+  const invokeAction = useCallback(
+    async (action: GQLAction) => {
```

Comment:
Don't use async nor useCallback, both are highlighting lifecycle issues

Reply:
ok for the async part, but for the `useCallback ` I'm don't know how to do without it.
The very most part of our _useXXXX_ hooks use the _useCallback_ mechanism.